myriad-scheduler/src/main/java/org/apache/myriad/configuration/MyriadConfiguration.java

```diff
@@ -197,47 +200,44 @@ public String getFrameworkRole() {
     return nmInstances;
   }
 
-  public Boolean isRebalancer() {
-    return rebalancer != null ? rebalancer : DEFAULT_REBALANCER;
+  public Boolean isRebalancerEnabled() {
+    return Optional.of(rebalancerEnabled).or(DEFAULT_REBALANCER_ENABLED);
```

Comment:
Argh, good catch! I added test cases to the MyriadConfigurationTest but
missed this null check.  Updating the code accordingly and test cases
accordingly.

Thanks!

--John

On Wed, May 11, 2016 at 10:52 PM, Darin <notifications@github.com> wrote:

> In
> myriad-scheduler/src/main/java/org/apache/myriad/configuration/MyriadConfiguration.java
> <https://github.com/apache/incubator-myriad/pull/71#discussion_r62958597>:
>
> > @@ -197,47 +200,44 @@ public String getFrameworkRole() {
> >      return nmInstances;
> >    }
> >
> > -  public Boolean isRebalancer() {
> > -    return rebalancer != null ? rebalancer : DEFAULT_REBALANCER;
> > +  public Boolean isRebalancerEnabled() {
> > +    return Optional.of(rebalancerEnabled).or(DEFAULT_REBALANCER_ENABLED);
>
> I think you want Optional.fromNullable no Optional.of, for the difference
> check out http://www.tutorialspoint.com/guava/guava_optional_class.htm.
> In fact, here's the method:
>
>   public static <T> Optional<T> of(T reference) {
>     return new Present<T>(checkNotNull(reference));
>   }
>
> checkNotNull is a guava precondition which will though an exception.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-myriad/pull/71/files/9bb1e6b7514d58b5cb128bd3e23bec0a03f55aaa#r62958597>
>